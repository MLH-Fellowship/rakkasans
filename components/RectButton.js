import React from "react";
import { Text, StyleSheet, TouchableOpacity, Image } from "react-native";

import Colors from "../constants/Colors";
import Dimensions from "../constants/Dimensions";

export default function RectButton({
  text,
  onPress,
  color = "primary",
  icons,
}) {
  return (
    <TouchableOpacity
      style={[styles.button, { backgroundColor: Colors[color] }]}
      onPress={onPress}
    >
      {icons === true && (
        <Image
          source={require("../assets/images/Torri.png")}
          style={styles.leftIcon}
        />
      )}
      <Text style={[styles.text, Dimensions.font]}>{text}</Text>
      {icons === true && (
        <Image
          source={require("../assets/images/Torri.png")}
          style={styles.rightIcon}
        />
      )}
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: Colors.primary,
    borderRadius: 5,
    justifyContent: "center",
    alignItems: "center",
    padding: 10,
    width: "100%",
    marginVertical: 10,
    flexDirection: "row",
  },
  text: {
    marginTop: 7,
    marginHorizontal: 30,
    fontSize: 18,
    color: Colors.white,
    backgroundColor: "transparent",
  },
  leftIcon: {
    width: 20,
    height: 20,
    marginLeft: 30,
  },
  rightIcon: {
    width: 20,
    height: 20,
    marginRight: 30,
  },
});
