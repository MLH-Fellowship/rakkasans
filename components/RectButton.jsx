import React from 'react';
import {
  Text, StyleSheet, TouchableOpacity, Image,
} from 'react-native';

import Colors from '../constants/Colors';
import Dimensions from '../constants/Dimensions';

import Torri from '../assets/images/Torri.png';

export default function RectButton({
  text,
  onPress,
  color = 'primary',
  icons,
}) {
  const styles = StyleSheet.create({
    button: {
      backgroundColor: Colors.primary,
      borderRadius: 5,
      justifyContent: 'center',
      alignItems: 'center',
      padding: 10,
      width: '100%',
      marginVertical: 10,
      flexDirection: 'row',
    },
    text: {
      marginTop: 7,
      marginHorizontal: 30,
      fontSize: 18,
      color: Colors.white,
      backgroundColor: 'transparent',
    },
    leftIcon: {
      width: 20,
      height: 20,
      marginLeft: 30,
    },
    rightIcon: {
      width: 20,
      height: 20,
      marginRight: 30,
    },
  });
  return (
    <TouchableOpacity
      style={[styles.button, { backgroundColor: Colors[color] }]}
      onPress={onPress}
    >
      {icons === true && (
        <Image
          source={Torri}
          style={styles.leftIcon}
        />
      )}
      <Text style={[styles.text, Dimensions.font]}>{text}</Text>
      {icons === true && (
        <Image
          source={Torri}
          style={styles.rightIcon}
        />
      )}
    </TouchableOpacity>
  );
}
