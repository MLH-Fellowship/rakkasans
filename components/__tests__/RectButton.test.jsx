import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { Image, Text } from 'react-native';
import renderer from 'react-test-renderer';

import RectButton from '../RectButton';

import Torri from '../../assets/images/Torri.png';

jest.useFakeTimers();
configure({ adapter: new Adapter() });

describe('RectButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RectButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with an image', () => {
    const wrapper = shallow(
      <RectButton>
        <Image src={Torri} />
        <Text>This is a test</Text>
        <Image src={Torri} />
      </RectButton>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('handles onPress', () => {
    const onPress = jest.fn();

    const wrapper = shallow(
      <RectButton onPress={onPress}>
        <Image src={Torri} />
        <Text>This is a test</Text>
        <Image src={Torri} />
      </RectButton>,
    );

    wrapper
      .find('TouchableOpacity')
      .first()
      .props()
      .onPress();

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
