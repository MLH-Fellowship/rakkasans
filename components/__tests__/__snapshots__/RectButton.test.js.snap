// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RectButton renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#020a41",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#ffffff",
          "fontSize": 18,
          "marginHorizontal": 30,
          "marginTop": 7,
        },
        Object {
          "fontFamily": "andale-mono",
        },
      ]
    }
  />
</View>
`;

exports[`RectButton renders correctly with an image 1`] = `ShallowWrapper {}`;
